Core/src/org/sleuthkit/autopsy/casemodule/Case.java

```diff
@@ -223,8 +205,16 @@ private Case(String name, String number, String examiner, String configFilePath,
         this.caseType = type;
         this.db = db;
         this.services = new Services(db);
-        this.eventSubscriber = new EventSubscriber();
-        this.messenger = new Messenger(this.name, eventPublisher);
+        if (CaseType.MULTI_USER_CASE == this.caseType) {
+            try {
+                eventPublisher.openRemoteEventChannel(name);
```

Comment:
Actually, the AutopsyEventPublisher for the Case class is static - this allows subscribers to subscribe once, instead of every time a new case is opened. So there actually is no leaking of a partially constructed object happening. However, some Case events are published in static methods, others are published in instance methods, and some are published by instance methods that call the static changeCase() method, so it is all rather messy and confusing. Originally, I had the calls to RemoteEventListener.openEventChannel and closeEventChannel() in the static changeCase() method. I will put them back there, instead of in the constructor/int()/closeCase() troika of instance methods. All roads lead to changeCase().

Note that I will do this in the next pull request for this feature, along with other work on the Case class and its events.